jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3ViewerPanel.kt

```diff
@@ -34,7 +35,13 @@ class S3ViewerPanel(disposable: Disposable, private val project: Project, privat
     private val rootNode: S3TreeDirectoryNode = S3TreeDirectoryNode(virtualBucket, null, "")
 
     init {
-        val structureTreeModel: StructureTreeModel<SimpleTreeStructure> = StructureTreeModel(SimpleTreeStructure.Impl(rootNode), disposable)
+        val structureTreeModel: StructureTreeModel<SimpleTreeStructure> = StructureTreeModel(
+            SimpleTreeStructure.Impl(rootNode),
+            null,
+            // TODO this has a concurrency of 1, do we want to adjust this?
+            Invoker.Background(disposable),
```

Comment:
If this is a concurrency of 1, use `Invoker.forBackgroundThreadWithoutReadAction`

Reply:
Updated